packages/anoncreds-rs/src/services/AnonCredsRsHolderService.ts

```diff
@@ -89,45 +96,50 @@ export class AnonCredsRsHolderService implements AnonCredsHolderService {
           retrievedCredentials.set(attribute.credentialId, credentialRecord)
         }
 
-        const credential = Credential.load(JSON.stringify(credentialRecord.credential))
-
-        const revocationRegistryDefinitionId = credential.revocationRegistryId
-        const revocationRegistryIndex = credential.revocationRegistryIndex
+        const revocationRegistryDefinitionId = credentialRecord.credential.rev_reg_id
+        const revocationRegistryIndex = credentialRecord.credentialRevocationId
 
         // TODO: Check if credential has a revocation registry id (check response from anoncreds-rs API, as it is
         // sending back a mandatory string in Credential.revocationRegistryId)
         const timestamp = attribute.timestamp
 
-        let revocationState
-        if (timestamp) {
-          if (revocationRegistryIndex) {
-            if (!options.revocationRegistries[revocationRegistryDefinitionId]) {
-              throw new AnonCredsRsError(`Revocation Registry ${revocationRegistryDefinitionId} not found`)
-            }
+        let revocationState: CredentialRevocationState | undefined
+        let revocationRegistryDefinition: RevocationRegistryDefinition | undefined
+        try {
+          if (timestamp) {
+            if (revocationRegistryIndex && revocationRegistryDefinitionId) {
```

Comment:
It's a bit difficult to read, but can we combine these? It's getting a bit deep here with the nesting.